File: configs/vim/python3/snippets/go.py
import os.path


def pkg_decl(snip, path):
    pkg = os.path.basename(os.path.dirname(os.path.realpath(path)))
    file_name = snip.basename
    if file_name.endswith("_test") and "internal" not in file_name:
        pkg = f"{pkg}_test"
    snip.rv = f"package {pkg}"

File: modules/neovim/lua/dotfiles/settings/vsnip/go.lua
local M = {}

function M.pkg_name()
    local file_name = vim.fn.expand("%:t:r")
    local dir_name = vim.fn.expand("%:h:t")

    if vim.endswith(file_name, "_test") and not vim.endswith(file_name, "_internal_test") then
        return dir_name .. "_test"
    end
    return dir_name
end

return M
